spring-integration-xmpp/src/main/java/org/springframework/integration/xmpp/inbound/ChatMessageListeningEndpoint.java

```diff
@@ -138,8 +140,17 @@ public void processPacket(final Stanza packet) {
 				Object messageBody = xmppMessage.getBody();
 
 				if (ChatMessageListeningEndpoint.this.payloadExpression != null) {
+					EvaluationContext evaluationContext = ChatMessageListeningEndpoint.this.evaluationContext;
+
+					List<ExtensionElement> extensions = xmppMessage.getExtensions();
+					if (extensions.size() == 1) {
+						ExtensionElement extension = extensions.get(0);
+						evaluationContext = ExpressionUtils.createStandardEvaluationContext(getBeanFactory());
+						evaluationContext.setVariable("extension", extension);
+					}
```

Comment:
That will be the same like `extensions`.
That variable isn't so useful.  :smile: 

See the diff for test-case.
With the single `extension` we can avoid to use its `namespace` and/or `element`.